llvm/lib/Transforms/Vectorize/VPlan.h

```diff
@@ -3597,6 +3656,11 @@ class VPlan {
   /// been modeled in VPlan directly.
   DenseMap<const SCEV *, VPValue *> SCEVToExpansion;
 
+  /// Stores the set of reduction exit instructions that will be scaled to
+  /// a smaller VF in this plan via partial reductions.
+  DenseMap<const Instruction *, PartialReductionChain>
```

Comment:
Does this need to be part of the VPlan class? Ideally all necessary information should be encoded in recipes. If it is just needed for construction VPRecipeBuilder would be a better place.